app/app/private-cloud/billings/all/page.tsx

```diff
@@ -0,0 +1,93 @@
+'use client';
+
+import { Button } from '@mantine/core';
+import { MonthPickerInput } from '@mantine/dates';
+import { useQuery } from '@tanstack/react-query';
+import { format } from 'date-fns';
+import { useState } from 'react';
+import { useSnapshot } from 'valtio/react';
+import Table from '@/components/generic/table/Table';
+import { GlobalPermissions } from '@/constants';
+import createClientPage from '@/core/client-page';
+import { getPrivateCloudAdminMonthlyCosts } from '@/services/backend/admin';
+import AdminCostTable from './AdminCostTable';
+import { pageState } from './state';
+
+const billingPage = createClientPage({
+  permissions: [GlobalPermissions.ViewPrivateCloudBilling],
+  fallbackUrl: 'login?callbackUrl=/home',
+});
+
+export default billingPage(({ session }) => {
+  if (!session?.previews.costRecovery) return null;
```

Comment:
this logic should be go after any `state` related logic like `useSnapshot`, `useQuery` to avoid runtime error